lib/default.nix

```diff
@@ -67,4 +67,50 @@ rec {
         mkDerivation = wrapFunctionWithChecks prev.stdenv.mkDerivation namePositions check;
       };
     };
+
+    # Creates an overlay that replaces buildPythonPackage with a function that,
+    # for packages with locations of name attribute matching one of the namePositions,
+    # checks the attribute set passed as argument
+    checkBuildPythonPackageFor =
+      check:
+    {
+      namePositions
+      , ...
+    }:
+    final:
+    prev:
+
+    let
+      pythonPackageSetNames = [
+        "python"
```

Comment:
It would be better to generate this list automatically.

Reply:
Replacing this expression with `pythonPackageSetNames = lib.attrNames prev.pythonInterpreters;` -- which seems like it absolutely should work(!) instead gives an infinite recursion error.